cpp_exercises/day3/phoneBook/phonebook.cpp

```diff
@@ -0,0 +1,18 @@
+#include "phonebook.h"
+
+PhoneBook::PhoneBook()
+{
+
+}
+
+void PhoneBook::append(const Contact& contact) {
+    m_contacts.push_back(contact);
+}
+
+const Contact* PhoneBook::find(const std::string& name) {
+    for(const Contact& contact: m_contacts) {
```

Comment:
Penso che l'idea di questa funzione in questo esercizio sia di spingerti a usare la libreria standard.
In questo caso la libreria standard ci viene incontro con [std::find_if](https://en.cppreference.com/w/cpp/algorithm/find) che ritorna un puntatore all'oggetto trovato o a `m_contacts.end()` se non ha trovato un match.
Se avessi un operatore di uguaglianza tra la classe Contact e una `std::string` basterebbe usare `std::find`, ma valutiamo prima se abbia un senso in base alle altre specifiche.